services/plugin/src/Plugin.tsx

```diff
@@ -47,10 +47,23 @@ export const Plugin = ({
 
     const [communicationReceived, setCommunicationReceived] =
         useState<boolean>(false)
+    const [prevCommunicationReceived, setPrevCommunicationReceived] =
+        useState<boolean>(false)
 
     const [inErrorState, setInErrorState] = useState<boolean>(false)
 
     useEffect(() => {
+        setCommunicationReceived(false)
+    }, [pluginEntryPoint])
+
+    useEffect(() => {
+        // if communicationReceived switches from false to true, the props have been sent
+        const prevCommunication = prevCommunicationReceived
+        setPrevCommunicationReceived(communicationReceived)
+        if (prevCommunication === false && communicationReceived === true) {
```

Comment:
this is done so that when communicationReceived toggles from false to true (after successful send of props), the props are not sent an additional time